core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -139,6 +154,7 @@ KOKKOS_INLINE_FUNCTION auto mapping_from_array_layout_impl(
         if (product_of_dimensions != layout.stride)
           Kokkos::abort(
               "Invalid conversion from LayoutRight to layout_right_padded");
+        return MappingType();
```

Comment:
Shouldn't we pass the extents to construct the `layout_right_padded` here ?

Reply:
yeah, this looks like a hack to me

Reply:
Actually this branch probably just needs to static_assert. This conversion doesn't really work because it turns out LayoutRight and layout_right_padded work differently.

Reply:
then do something like
```C++
static_assert(std::is_void_v<ArrayLayout>);

Reply:
Done